one_fm/api/mobile/Leave_application.py

```diff
@@ -134,4 +133,60 @@ def response(message, data, status_code):
      frappe.local.response["message"] = message
      frappe.local.response["data_obj"] = data
      frappe.local.response["http_status_code"] = status_code
-     return
\ No newline at end of file
+     return
+
+@frappe.whitelist()
+def get_leave_approver(employee):
+    """
+    This function fetches the leave approver for a given employee.
+    The leave approver is fetched  either Report_to or Leave Approver. 
+    But, if both don't exist, Operation manager is the Leave Approver.
+
+    Params: ERP Employee ID
+
+    Return: User ID of Leave Approver
+
+    """
+    approver = None
+    #check if employee has leave approver or report to assigned in the employee doctype
+    leave_approver, report_to = frappe.db.get_value("Employee", employee, ["leave_approver", "reports_to"])
+    if not leave_approver:
+        if report_to:
+            approver = frappe.db.get_value('Employee', {'name': report_to}, ['user_id'])
+        else:
+            #if not, return the 'Operational Manager' as the leave approver. But, check if employee himself is not a leave manager.
+            operation_manager = frappe.db.get_value('Employee', {'Designation': "Operation Manager"}, ['name','user_id'])
```

Comment:
I guess filtering by designation is not a good practice. What if multiple employees have the same designation, then who will approve the leave?